File: ZCO14003.py
try:
    N = int(input())
    L = []
    for _ in range(N):
        x = int(input())
        L.append(x)
    L.sort()
    P = []
    for i in L:
        P.append(i * N)
        N -= 1
    print(max(P))
except:
    pass

File: ZCO14003.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    long long n;
    cin >> n;
    vector<int> prices(n);
    for(int i = 0; i < n; i++)
        cin >> prices[i];
    sort(prices.begin(), prices.end());
    long long maximum = 0;
    for(int i = 0; i < n; i++)
        maximum = max(maximum, (long long)prices[i]*(n-i));
    cout << maximum << endl;
	return 0;
}
